sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +40,352 @@ class ConfigurationSetting(Model):
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    key_prefix = ".appconfig.featureflag/"
+    _feature_flag_content_type = (
+        "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+    )
+    kind = "FeatureFlag"
+
+    def __init__(self, key, enabled, filters=[], **kwargs):  # pylint: disable=dangerous-default-value
+        # type: (str, bool, Optional[List[Dict[str, Any]]]) -> None
+        super(FeatureFlagConfigurationSetting, self).__init__(**kwargs)
+        if not key.startswith(self.key_prefix):
+            key = self.key_prefix + key
+        self.key = key
+        self.label = kwargs.get("label", None)
+        self.content_type = kwargs.get("content_type", self._feature_flag_content_type)
+        self.last_modified = kwargs.get("last_modified", None)
+        self.tags = kwargs.get("tags", {})
+        self.read_only = kwargs.get("read_only", None)
+        self.etag = kwargs.get("etag", None)
+        self.description = kwargs.get("description", None)
+        self.display_name = kwargs.get("display_name", None)
+        self._value = kwargs.get("value", {"enabled": enabled, "conditions": {"client_filters": filters}})
+
+    def _validate(self):
+        # type: () -> None
+        if not self.key.startswith(self.key_prefix):
+            raise ValueError("All FeatureFlagConfigurationSettings should be prefixed with {}.".format(self.key_prefix))
+        if not (self._value is None or isinstance(self._value, dict)):
+            raise ValueError("Expect 'value' to be a dictionary.")
+
+    @property
+    def enabled(self):
+        # type: () -> Union[None, bool]
+        self._validate()
+        if self._value is None or "enabled" not in self._value:
+            return None
+        return self._value["enabled"]
+
+    @enabled.setter
+    def enabled(self, new_value):
+        # type: (bool) -> bool
+        self._validate()
+        if self._value is None:
+            self._value = {}
+        self._value["enabled"] = new_value
+
+    @property
+    def filters(self):
+        # type: () -> Union[None, List[Any]]
+        self._validate()
+        if self._value is None:
+            return None
+        return self._value["conditions"]["client_filters"]
+
+    @filters.setter
+    def filters(self, new_filters):
+        # type: (List[Dict[str, Any]]) -> None
+        self._validate()
+        if self._value is None:
+            self._value = {}
+        try:
+            self._value["conditions"]["client_filters"] = new_filters
+        except KeyError:
+            self._value["conditions"] = {
+                "client_filters": new_filters
+            }
+
+    @property
+    def value(self):
+        # type: () -> Dict[str, Any]
+        self._validate()
```

Comment:
We don't need validate() here. value is a property of basic config setting. It is always valid.